servers/jax-rs-tests/pom.xml

```diff
@@ -99,6 +99,12 @@
       <artifactId>nessie-gc-base</artifactId>
       <version>${project.version}</version>
     </dependency>
+    <dependency>
+      <groupId>org.projectnessie</groupId>
+      <artifactId>nessie-spark-extensions</artifactId>
```

Comment:
Because usually the Nessie GC is used with Nessie + spark + iceberg.
Nessie only works with spark3.1 with Iceberg[1], so used the same environment settings here.

[1]As per compatibility doc
https://github.com/projectnessie/nessie#compatibility